pkg/rules/manager.go

```diff
@@ -144,9 +144,10 @@ func NewManager(
 	return m
 }
 
+// Run is non blocking, in opposite to TSDB manager, which is blocking.
 func (m *Manager) Run() {
 	for _, mgr := range m.mgrs {
-		mgr.Run()
+		go mgr.Run()
```

Comment:
How about using a rungroup or errgroup for these routine? At least we should have more visibility on `Run()` in case of failures, maybe log the something in case of failure?

Reply:
I checked the internals and the inside Prometheus thingy it's quite simple so run group would not help us much here. Something to do in future though (: